src/merge_data.rs

```diff
@@ -192,6 +192,16 @@ impl MergeDeferredDataSource {
 }
 
 impl DeferredDataSource for MergeDeferredDataSource {
+    fn fetch_description(&self) -> DataSourceDescription {
+        DataSourceDescription {
+            source_locator: self.data_sources.iter().fold(Vec::new(), |acc, x| {
```

Comment:
Same comment as above, I think this can be done more simply with `flatten`.

Reply:
https://github.com/StanfordLegion/prof-viewer/pull/36/commits/20ba2e8e877f7d44a34eb35958ed0eb07334d635